compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -297,13 +297,13 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
             onConflictExpr = null;
         } else {
             BType refType = Types.getReferredType(queryExpr.getBType());
-            if (isXml(refType)) {
+            if (isXml(types.getSafeType(refType, true, true))) {
```

Comment:
Get the safe type commonly for both cases

Reply:
Added. 35ec75614ab6aa67688f321ffeb8b3c19441b285